aa/data.py

```diff
@@ -104,6 +142,20 @@ def __init__(self, pv, values, timestamps, severities):
         self._values = values
         self._timestamps = timestamps
         self._severities = severities
+        self._enum_options = enum_options
+        self._enum_strings = numpy.empty_like(values, dtype="U100")
+        if enum_options:
+            self._populate_enum_strings()
+
+    def _populate_enum_strings(self) -> None:
```

Comment:
Do this at construction or when requested? Debatable.

Reply:
Could additionally re-calculate if enum_options is changed. I wouldn't want to do it on every get of enum_strings as this isn't consistent with what we do with the values, i.e. have them in a static array for quick access.

Reply:
I'm not sure I agree here. 

Firstly I think the data structures should be (implicitly) immutable, so I don't think we have to worry about anything changing.

I tend to think that the raw data is <numbers> plus <enum dict> and storing the <enum strings> is a bit redundant. This is theoretical: in my case if you request enum strings multiple times you lose CPU cycles; in yours you lose memory.

In practice we will never see any performance problems from either so this is a style question. I absolutely don't mind which way you pick, just discussing pros and cons.

Reply:
I went for your suggestion in the end, because it was easier to manage and pass around just the options, and more understandable what is going on with it in the code at each point. I learned about vectorizing functions for numpy, hopefully I've done it right in this case.